src/test/java/com/squareup/javapoet/TypeSpecTest.java

```diff
@@ -441,6 +441,68 @@ private boolean isJava8() {
         + "}\n");
   }
 
+  /** https://github.com/square/javapoet/issues/253 */
+  @Test public void enumWithAnnotatedValues() throws Exception {
+    TypeSpec roshambo = TypeSpec.enumBuilder("Roshambo")
+        .addModifiers(Modifier.PUBLIC)
+        .addEnumConstant("ROCK", TypeSpec.anonymousClassBuilder("")
+            .addJavadoc("Avalanche!\n")
+            .addAnnotation(ClassName.get("javax.annotation", "Nullable"))
```

Comment:
This is an awkward example because this annotation can't actually go here. Maybe use `@Deprecated` instead?